File: java/src/array/ArrayQueue.java
package array;

public class ArrayQueue<E> {
    private E[] array;

    private int read;

    private int write;

    public ArrayQueue() {
        this.array = (E[]) new Object[5];
    }

    public String toString() {
        int i = read;

        StringBuffer sb = new StringBuffer();
        sb.append("[");
        while(array[i] != null) {
            sb.append(array[i]);
            sb.append(",");

            i = increment(i);
        }
        sb.deleteCharAt(sb.length() - 1);
        sb.append("]");

        return sb.toString();
    }

    public void enQueue(E input) {
        if (canEnQueue()) {
            array[write] = input;
            write = increment(write);
        } else {
            throw new IllegalStateException();
        }
    }

    private boolean canEnQueue() {
        if ((read == 0 ? array.length - 1 : read - 1) == write)
            return false;
        else
            return true;
    }

    public E deQueue() {
        if (empty())
            return null;

        E del = array[read];
        array[read] = null;
        read = increment(read);
        return del;
    }

    private int increment(int i) {
        return (i == array.length - 1) ? 0 : i + 1;
    }

    public boolean empty() {
        return read == write;
    }
}

File: python/array/arrayQueue.py
class ArrayQueue:
    arr = [None]*5
    write = 0
    read = 0
    
    def __init__(self):
        arr = [None]*5
        self.write = 0
        self.read = 0

    def toString(self):
        result = [str(d) for d in self.arr if d != None]
        return ','.join(result)

    def enqueue(self, input):
        nextIn = self.__nextWrite()

        if nextIn == self.read:
            raise IndexError
        else:
            self.arr[self.write] = input
            self.write = nextIn
    
    def __nextWrite(self):
        return 0 if self.write == len(self.arr) - 1 else self.write + 1

    def dequeue(self):
        if self.write == self.read:
            return None
        result = self.arr[self.read]
        self.arr[self.read] = None
        self.read = 0 if self.read == len(self.arr) - 1 else self.read + 1
        
        return result

    def empty(self):
        return self.read == self.write
